teammapper-frontend/mmp/src/map/handlers/nodes.ts

```diff
@@ -640,6 +641,77 @@ export default class Nodes {
         }
     }
 
+    /**
+     * Proactively apply coordinates to all nodes that are missing coordinates
+     * This method is very similar to calculateCoordinates, but it's fully self-contained and only applicable to MapSnapshots.
+     * Adding this functionality to calculateCoordinates would break a lot of things, as Node and ExportNodeProperties are not the same.
+     * @param {MapSnapshot} map
+     * @returns {MapSnapshot} map - Return the map snapshot with properly applied coordinates
+     */
+    public applyCoordinatesToMapSnapshot(map: MapSnapshot): MapSnapshot {
+        const getOrientation = (node: ExportNodeProperties, rootNode: ExportNodeProperties) => node.coordinates?.x < rootNode.coordinates?.x
+        const getLowerNode = (nodes: ExportNodeProperties[]) => {
+            let tmp = nodes[0].coordinates?.y || 0
```

Comment:
This can be improved. Don't use let as a result value and map to find something. There is also `reduce`, `select` etc. to do this. I always try to avoid let and only use const. If you really have to, then map is the wrong iteration function.

Reply:
I've refactored the original method and made some adjustments so there's no longer a special version inside `applyCoordinatesToMapSnapshot`, which means the other methods also benefit from the refactoring. Same applies to `getOrientation`.